docs/tutorials/tabular_prediction/tabular-custom-model.md

```diff
@@ -0,0 +1,374 @@
+test_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/test.csv')  # another Pandas DataFrame
+label = 'class'  # specifies which column do we want to predict
+train_data = train_data.sample(n=1000, random_state=0)  # subsample for faster demo
+
+train_data.head(5)
+```
+
+## Training a custom model without TabularPredictor
+
+Below we will demonstrate how to train the model outside [TabularPredictor](../../api/autogluon.predictor.html#module-0). This is useful for debugging and minimizing the amount of code you need to understand while implementing the model.
+
+This process is similar to what happens internally when calling fit on `TabularPredictor`, but is simplified and minimal.
+
+If the data was already cleaned (all numeric), then we could call fit directly with the data, but the adult dataset is not.
+
+### Clean labels
+
+The first step to making the input data as valid input to the model is to clean the labels.
+
+Currently, they are strings, but we need to convert them to numeric values (0 and 1) for binary classification.
+
+Luckily, AutoGluon already implements logic to both detect that this is binary classification (via `infer_problem_type`), and a converter to map the labels to 0 and 1 (`LabelCleaner`):
+
+```{.python .input}
+# Separate features and labels
+X = train_data.drop(columns=[label])
+y = train_data[label]
+X_test = test_data.drop(columns=[label])
+y_test = test_data[label]
+
+from autogluon.core.data import LabelCleaner
+from autogluon.core.utils import infer_problem_type
+# Construct a LabelCleaner to neatly convert labels to float/integers during model training/inference, can also use to inverse_transform back to original.
+problem_type = infer_problem_type(y=y)  # Infer problem type (or else specify directly)
+label_cleaner = LabelCleaner.construct(problem_type=problem_type, y=y)
+y_clean = label_cleaner.transform(y)
+
+print(f'Labels cleaned: {label_cleaner.inv_map}')
+print(f'inferred problem type as: {problem_type}')
+print('Cleaned label values:')
+y_clean.head(5)
+```
+
+### Clean features
+
+Next, we need to clean the features. Currently, features like 'workclass' are object dtypes (strings), but we actually want to use them as categorical features. Most models won't accept string inputs, so we need to convert the strings to numbers.
+
+AutoGluon contains an entire module dedicated to cleaning, transforming, and generating features called [autogluon.features](../../api/autogluon.features.html). Here we will use the same feature generator used internally by `TabularPredictor` to convert the object dtypes to categorical and minimize memory usage.
+
+```{.python .input}
+from autogluon.features.generators import AutoMLPipelineFeatureGenerator
+
+feature_generator = AutoMLPipelineFeatureGenerator()
+X_clean = feature_generator.fit_transform(X)
+
+X_clean.head(5)
+```
+
```

Comment:
may be helpful to clarify what AutoMLPipelineFeatureGenerator does not do for you, but many models might like to have (eg. rescaling/imputation of numerical features)

Reply:
added